test/unit/gateways/global_collect_test.rb

```diff
@@ -71,6 +92,130 @@ def test_successful_purchase_with_requires_approval_true
     end.respond_with(successful_authorize_response, successful_capture_response)
   end
 
+  def test_purchase_request_with_google_pay
+    stub_comms do
+      @gateway.purchase(@accepted_amount, @google_pay_network_token)
+    end.check_request(skip_response: true) do |_endpoint, data, _headers|
+      assert_equal '320', JSON.parse(data)['mobilePaymentMethodSpecificInput']['paymentProductId']
+    end
+  end
+
+  def test_purchase_request_with_google_pay_pan_only
+    stub_comms do
+      @gateway.purchase(@accepted_amount, @google_pay_pan_only, @options.merge(customer: 'GP1234ID', google_pay_pan_only: true))
+    end.check_request(skip_response: true) do |_endpoint, data, _headers|
+      assert_equal '320', JSON.parse(data)['mobilePaymentMethodSpecificInput']['paymentProductId']
+    end
+  end
+
+  def test_add_payment_for_credit_card
+    post = {}
+    options = {}
+    payment = @credit_card
+    @gateway.send('add_payment', post, payment, options)
+    assert_includes post.keys, 'cardPaymentMethodSpecificInput'
+    assert_equal post['cardPaymentMethodSpecificInput']['paymentProductId'], '1'
+    assert_equal post['cardPaymentMethodSpecificInput']['authorizationMode'], 'FINAL_AUTHORIZATION'
+    assert_includes post['cardPaymentMethodSpecificInput'].keys, 'card'
+    assert_equal post['cardPaymentMethodSpecificInput']['card']['cvv'], '123'
+    assert_equal post['cardPaymentMethodSpecificInput']['card']['cardNumber'], '4567350000427977'
+    assert_equal post['cardPaymentMethodSpecificInput']['card']['expiryDate'], '0923'
```

Comment:
exp date checks are brittle since the default card is created dynamically

Reply:
Ok, I'll fix it